src/SFA.DAS.AdminService.Web/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,17 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using System.Linq;
+using System.Reflection;
+
+namespace SFA.DAS.AdminService.Web.Extensions
+{
+    public static class ServiceCollectionExtensions
+    {
+        public static void RegisterAllTypes<T>(this IServiceCollection services, Assembly[] assemblies,
+            ServiceLifetime lifetime = ServiceLifetime.Transient)
+        {
+            var typesFromAssemblies = assemblies.SelectMany(a => a.DefinedTypes.Where(x => x.GetInterfaces().Contains(typeof(T))));
+            foreach (var type in typesFromAssemblies)
```

Comment:
We had to change this line slightly to `foreach (var type in typesFromAssemblies.Where(t=>!t.IsAbstract))` because we had an abstract class that implemented IValidator without a default ctor.  YMMV. 👍

Reply:
We're not using this yet, but I've updated it anyway, for future use